FixDevV3Blobs/FixDevV3Blobs/Program.cs

```diff
@@ -0,0 +1,279 @@
+        {
+            var stopwatch = Stopwatch.StartNew();
+            var count = 0;
+            var skipped = 0;
+            var blobCount = blobList.Count;
+
+            var tasks = Enumerable.Range(1, MaxTasks).Select(async _ => 
+            {
+                await Task.Yield();
+                while (blobList.TryTake(out var blob))
+                {
+                    var curCount = Interlocked.Increment(ref count);
+                    if (Processed(blob))
+                    {
+                        Interlocked.Increment(ref skipped);
+                        continue;
+                    }
+                    int attempt = 0;
+                    do
+                    {
+                        try
+                        {
+                            await ProcessBlobAsync(blob, search, replace);
+                            break;
+                        }
+                        catch (UnsupportedContentEncodingException e)
+                        {
+                            Log($"{blob.Uri.AbsoluteUri}: {e.Message}");
+                            attempt = MaxRetries;
+                        }
+                        catch (Exception e)
+                        {
+                            Log($"Got exception while processing {blob.Uri.AbsoluteUri}: {e}");
+                            await Task.Delay(TimeSpan.FromSeconds(10));
+                        }
+                    } while (++attempt < MaxRetries);
+                    if (attempt == MaxRetries)
+                    {
+                        LogWarning($"Did not process {blob.Uri.AbsoluteUri}");
+                    }
+                    if ((curCount % 1000) == 0)
+                    {
+                        Log(
+                            "{0}/{1} (skipped: {2}) {3}, ETA: {4} (uncompressed: {5}/{6}, gzip: {7}/{8})",
+                            curCount,
+                            blobCount,
+                            skipped,
+                            stopwatch.Elapsed,
+                            TimeSpan.FromSeconds(((double)blobCount - curCount) / curCount * stopwatch.Elapsed.TotalSeconds),
+                            readUncompressedBlobs,
+                            writtenUncompressedBlobs,
+                            readGzipBlobs,
+                            writtenGzipBlobs);
+                    }
+                }
+            });
+            await Task.WhenAll(tasks);
+            stopwatch.Stop();
+            return stopwatch;
+        }
+
+        private static bool Processed(CloudBlockBlob blob)
+        {
+            return ProcessedBlobs.Contains(blob.Uri.AbsoluteUri);
+        }
+
+        private static async Task ProcessBlobAsync(CloudBlockBlob blob, string search, string replace)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            var content = await GetBlobContentAsync(blob);
+            stopwatch.Stop();
+
+            //SaveLocalContent("input", blob, content);
```

Comment:
Please remove if not needed. Same for other commented code below.

Reply:
It is useful for testing to make sure the changes applied are the ones that desired. Wrapped with `if` isntead of commenting.